include/cpr/session.h

```diff
@@ -45,11 +46,15 @@ class Interceptor;
 class MultiPerform;
 
 class Session : public std::enable_shared_from_this<Session> {
+  private:
+    curl_slist * curlSlist_ = nullptr;
   public:
     Session();
     Session(const Session& other) = delete;
 
-    ~Session() = default;
+    ~Session() {
+      curl_slist_free_all(curlSlist_);
```

Comment:
This is probably something we won't get around.
What one could do is create a wrapper type around `curl_slist` (e.g. `SList`), following the RAII pattern. In my eyes this would make more sense here.